### PR TITLE
Implement new copyright compliance and PIA notices

### PR DESCRIPTION
**Pull request by grahamb**
This is a squashed version of #201. Please vote.

Replace the old SFU Copyright Compliance and PIA notices
with React.js components.

This is a combination of 30 commits.
The first commit's message is:
initial commit of SFUCopyrightComplianceNotice React Component

This is the 2nd commit message:
initial commit of SFUCopyrightComplianceNoticeMoreInfo component

This is the 3rd commit message:
add sfu-ic-wizard-box__message-text__copyright_compliance to sfu.css

This is the 4th commit message:
add functionality to show more info

This is the 5th commit message:
insert SFUCopyrightComplianceNotice into InfoFrame

This is the 6th commit message:
remove handler for showing confirm in the course wizard

This is the 7th commit message:
add SFUCopyrightComplianceNoticeModalDialog component

This is the 8th commit message:
create copyright_notice_modal_dialog requirejs module

handles attaching a click handler to an element to call SFUCopyrightComplianceModalDialog

This is the 9th commit message:
require copyright_notice_modal_dialog in sfu.js and attach click handler

This is the 10th commit message:
clean up code in sfu.js

On a course page:
 - if the course is published, do nothing; otherwise
 - disable the publish button
 - load the copyright_notice_modal_dialog bundle async
    - bundle re-enables the button and attaches the click handler to call SFUCopyrightComplianceNoticeModalDialog

This is the 11th commit message:
reuse the existing div#wizard_box element for the modal

This is the 12th commit message:
re-enable the submit button

This is the 13th commit message:
Clean up SFUCopyrightComplianceNoticeModalDialog

add propTypes

read from props in getInitialState

code cleanup

use componentWillReceiveProps to setState on subsequent invocations

e.g. when the dialog is cancelled and recalled

This is the 14th commit message:
remove external apps pia injection code from sfu.js

This is the 15th commit message:
add icon style overrides for external apps PIA notice

This is the 16th commit message:
add SFUPrivacyNoticeExternaApps component

This is the 17th commit message:
include SFUPrivacyNoticeExternalApps component in external_apps/Header

This is the 18th commit message:
rename SFUPrivacyNoticeExternalApps → SFUPrivacyNotice and genericize

This is the 19th commit message:
change css classes

This is the 20th commit message:
use renamed component and pass usage prop

This is the 21st commit message:
receive alertStyle via props

This is the 22nd commit message:
make .alert text darker

This is the 23rd commit message:
fix comment

This is the 24th commit message:
add google_docs_pia_notice module

This is the 25th commit message:
replace google docs pia notice with new react component

This is the 26th commit message:
remove redundant warning message

This is the 27th commit message:
add comment to top of sfu.js noting where the sfu-modules are located

Mostly for future-gnb.

This is the 28th commit message:
remove hard-coded copyright notices from course form views

This is the 29th commit message:
use react to render the copyright notice in the new course form

This is the 30th commit message:
remove copyright disclaimer from sfu_copyright plugin

### CODE REVIEW

**Comment by va7map** [comment]
:+1:

File: app/jsx/course_wizard/InfoFrame.jsx [hunk 1]
```diff
@@ -134,6 +147,7 @@ define([
                       <p className="ic-wizard-box__message-text">
                         {this.state.itemShown.text}
                       </p>
+                      {this.renderSFUCopyrightComplianceNotice()}
```
Comment (va7map) [inline app/jsx/course_wizard/InfoFrame.jsx]: Do we want to bother with `SFU MOD` here?

File: app/jsx/external_apps/components/Header.jsx [hunk 2]
```diff
@@ -31,7 +32,11 @@ define([
             <p>{I18n.t('Apps are an easy way to add new features to Canvas. They can be added to individual courses, or to all courses in an account. Once configured, you can link to them through course modules and create assignments for assessment tools.')}</p>
             <p dangerouslySetInnerHTML={{ __html: paragraph }}></p>
           </div>
+
+          <SFUPrivacyNotice alertStyle="alert" usage="external_apps" />
```
Comment (va7map) [inline app/jsx/external_apps/components/Header.jsx]: ... and here?

Comment (grahamb) [inline app/jsx/external_apps/components/Header.jsx]: I was going to, but getting a comment into JSX was giving me troubles so I took it out.

Comment (va7map) [inline app/jsx/external_apps/components/Header.jsx]: Oh well. At least the letters "SFU" are in there, so it's not too hard to spot/search for.

**Comment by grahamb** [comment]
I've added the SFU MOD comments to the JSX files (`{/* SFU MOD */}`). @GinnyGuo thumbed #201, so I'm going to merge this into dev and push to test.

**Pull request merged**